crates/red_knot/src/types.rs

```diff
@@ -37,40 +38,137 @@ impl TypeEnvironment {
     }
 }
 
+#[derive(Debug)]
 pub(crate) enum Type {
-    Class(ClassType),
+    // a specific function
     Function(FunctionType),
+    // the set of Python objects with this class in their MRO
+    Instance(ClassType),
```

Comment:
It's a bit confusing that the type is called `ClassType` but the variant is `Instance`. I think I would align the two.

Reply:
Or you could have something like

```suggestion
    Instance { instance_of: ClassType },
```

Reply:
Yeah, let me think about this a bit more. I think I like Alex's version. The inner struct is really a representation of a class,  but the type represents instances of that class.